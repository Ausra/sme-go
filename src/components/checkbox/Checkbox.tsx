import { FunctionComponent, ReactElement, useState } from "react";
import { Container, StyledCheckboxInput, StyledLabel } from "./style";

export interface CheckboxProps {
  dataTestId?: string;
  children?: string | ReactElement[];
  id: string;
  showChecked?: boolean;
  onChangeCallback?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const defaultTestId = "styled-checkbox-container";
export const checkboxTestId = "styled-checkbox-input";

const Checkbox: FunctionComponent<CheckboxProps> = ({
  dataTestId,
  children,
  id,
  showChecked,
  onChangeCallback,
}) => {
  const [checked, setChecked] = useState(false);
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setChecked(event.target.checked);
    onChangeCallback && onChangeCallback(event);
  };
  return (
    <Container data-testid={defaultTestId}>
      <StyledCheckboxInput
        data-testid={dataTestId || checkboxTestId}
        type="checkbox"
        id={id}
        checked={showChecked || checked}
        onChange={handleChange}
      />
      <StyledLabel htmlFor={id}>{children}</StyledLabel>
    </Container>
  );
};

export default Checkbox;
