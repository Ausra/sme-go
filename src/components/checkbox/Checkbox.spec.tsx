import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Checkbox, { checkboxTestId, defaultTestId } from "./Checkbox";
import "jest-styled-components";

describe("Checkbox Styles", () => {
  it("should render", () => {
    render(<Checkbox id="checkbox1" />);
    const checkbox = screen.getByTestId(defaultTestId);
    expect(checkbox).toMatchSnapshot();
  });
});

describe("Checkbox", () => {
  it("should render default Checkbox", () => {
    render(<Checkbox id="checkbox1" />);
    const checkbox = screen.getByTestId(defaultTestId);
    expect(checkbox).toBeInTheDocument();
  });
  it("should call onChangeCallback", async () => {
    const user = userEvent.setup();
    const onChangeCallback = jest.fn();
    render(<Checkbox id="checkbox1" onChangeCallback={onChangeCallback} />);
    const checkbox = screen.getByTestId(checkboxTestId);
    await user.click(checkbox);
    expect(onChangeCallback).toHaveBeenCalled();
  });
});
