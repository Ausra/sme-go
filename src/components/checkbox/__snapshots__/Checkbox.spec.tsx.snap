// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Styles should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  outline: 1px solid #CCCCCC;
  border: transparent;
}

.c2 {
  font-size: 12px;
}

<div
  class="c0"
  data-testid="styled-checkbox-container"
>
  <input
    class="c1"
    data-testid="styled-checkbox-input"
    id="checkbox1"
    type="checkbox"
  />
  <label
    class="c2"
    for="checkbox1"
  />
</div>
`;
